torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -137,18 +137,28 @@ static inline Variable applySlicing(
     variable_list& outIndices,
     bool is_tracing,
     const at::Device& self_device,
-    const IntArrayRef& self_sizes,
+    const c10::optional<IntArrayRef> & self_sizes,
     int64_t specified_dims) {
   int64_t size = PyTuple_GET_SIZE(index); // NOLINT(cppcoreguidelines-pro-type-cstyle-cast)
   int64_t dim = 0;
 
-  if (specified_dims > (int64_t)self_sizes.size()) {
-    throw IndexError("too many indices for tensor of dimension %d", (int)(self_sizes.size()));
+  // TODO: nested tensor does not have a size (yet) so for now we represent its size as null
```

Comment:
For repeat notes like this use a "tagging" style and write the note in a single location. For example https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/TensorIndexing.h#L365
```
// See NOTE [ Setting `disable_slice_optimization` when calling C++ tensor indexing functions from Python ]
static inline Tensor handleDimInMultiDimIndexing(
```

which is then defined in a single location https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/TensorIndexing.h#L469-L495